### PR TITLE
Add position to Player and test by printing coordinates

### PR DESCRIPTION
**Pull request by pottumuusi**
Add in order to have something to test in unit tests. Also required functionality for game.

### CODE REVIEW

File: include/actor.h [hunk 1]
```diff
@@ -3,11 +3,14 @@
 
 #include <string>
 
-class Actor {
+#include "locatable.h"
+
+class Actor : public Locatable {
```
Comment (pottumuusi) [inline include/actor.h:8]: Rationale:
All actors are to have a location in world.

File: include/actor.h [hunk 2]
```diff
@@ -3,11 +3,14 @@
...
 public:
-    Actor();
+    Actor() = delete;
     ~Actor();
     Actor(std::string);
+    Actor(std::string, const int _x, const int _y);
```
Comment (pottumuusi) [inline include/actor.h:13]: Enum type for x and y would make it much more difficult to accidentially set coordinates in wrong order.

Comment (pottumuusi) [reply include/actor.h:13]: Issue created for fixing this. (#18)

File: src/game.cxx [hunk 3]
```diff
@@ -12,13 +12,23 @@ Game::~Game() {
...
-    actors.push_back(player);
+    actors.push_back(player1);
+    actors.push_back(player2);
 }
 
 void Game::test_print() const {
     std::cout << "Name of actors[0] is: " << actors[0].getName() << std::endl;
+    std::cout << "Name of actors[1] is: " << actors[1].getName() << std::endl;
+    std::cout << "Coordinates of actors[1] is: ("
+        << actors[1].getPosition().getX()
```
Comment (pottumuusi) [inline src/game.cxx:30]: Should remove trainwreck

Comment (pottumuusi) [reply src/game.cxx:30]: Issue created (#19)

File: src/locatable.cxx [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+Locatable::~Locatable()
+{
+    std::cout << "Destructor of Locatable running" << std::endl;
+}
+
+Locatable::Locatable(const int _x, const int _y)
+{
+    std::cout << "Constructor (int, int) of Locatable running" << std::endl;
+    position = Position{_x, _y};
```
Comment (pottumuusi) [inline src/locatable.cxx]: This is fixed in next commit (02702cc)

File: src/actor.cxx [hunk 5]
```diff
@@ -2,19 +2,27 @@
...
-Actor::Actor() {
-    std::cout << "Default constructor of Actor running" << std::endl;
+Actor::~Actor()
+{
+    std::cout << "Destructor of Actor running" << std::endl;
 }
 
-Actor::~Actor() {
-    std::cout << "Destructor of Actor running" << std::endl;
+Actor::Actor(std::string _name)
```
Comment (pottumuusi) [inline src/actor.cxx:10]: Duplicated implementation. Call own more detailed constructor instead. Same issue in player.cxx. Fix in: (bb43fe9)

File: src/locatable.cxx [hunk 6]
```diff
@@ -0,0 +1,20 @@
+#include <iostream>
+
+#include "locatable.h"
+
+Locatable::~Locatable()
+{
+    std::cout << "Destructor of Locatable running" << std::endl;
```
Comment (pottumuusi) [inline src/locatable.cxx:7]: Remove constructor and destructor prints. Test with unit tests instead.

Comment (pottumuusi) [reply src/locatable.cxx:7]: Issue created (#20)

**Review by ghost — APPROVED** [review]

**Pull request merged**